hl7apy/mllp.py

```diff
@@ -47,50 +54,54 @@ def __str__(self):
 
 
 class _MLLPRequestHandler(StreamRequestHandler):
+    encoding = 'utf-8'
+
     def __init__(self, *args, **kwargs):
         StreamRequestHandler.__init__(self, *args, **kwargs)
 
     def setup(self):
-        self.sb = "\x0b"
-        self.eb = "\x1c"
-        self.cr = "\x0d"
-        self.validator = re.compile(self.sb + "(([^\r]+\r)*([^\r]+\r?))" + self.eb + self.cr)
+        self.sb = b"\x0b"
+        self.eb = b"\x1c"
+        self.cr = b"\x0d"
+        self.validator = re.compile(
+            self.sb.decode('ascii') + r"(([^\r]+\r)*([^\r]+\r?))" + self.eb.decode('ascii') + self.cr.decode('ascii'))
         self.handlers = self.server.handlers
         self.timeout = self.server.timeout
 
         StreamRequestHandler.setup(self)
 
     def handle(self):
-        end_seq = "{}{}".format(self.eb, self.cr)
+        end_seq = self.eb + self.cr
+        io = BytesIO()
         try:
-            line = self.request.recv(3)
+            io.write(self.request.recv(3))
         except socket.timeout:
             self.request.close()
             return
 
-        if line[0] != self.sb:  # First MLLP char
+        if io.getvalue()[:1] != self.sb:  # First MLLP char
             self.request.close()
             return
 
-        while line[-2:] != end_seq:
+        while io.getvalue()[-2:] != end_seq:
```

Comment:
I think this should be safe in both LATIN-1 (because each byte is also a character) and UTF-8 (since the bytes in `end_seq` are in the 0-127 range).